addons/l10n_it_edi_extended/tests/test_withholding.py

```diff
@@ -0,0 +1,282 @@
+                    **cls.withholding_sale_line,
+                    'name': name,
+                    'price_unit': price,
+                }) for (name, price) in cls.real_client_invoice_lines
+            ],
+        })
+
+        cls.pension_fund_tax_invoice = cls.env['account.move'].with_company(cls.company).create({
+            'move_type': 'out_invoice',
+            'company_id': cls.company.id,
+            'partner_id': cls.italian_partner_a.id,
+            'invoice_date': datetime.date(2022, 3, 24),
+            'invoice_line_ids': [
+                (0, 0, {
+                    **cls.pension_fund_sale_line,
+                    'name': name,
+                    'price_unit': price,
+                }) for (name, price) in cls.real_client_invoice_lines
+            ]
+        })
+
+        total = sum(x[1] for x in cls.real_client_invoice_lines)
+        tax_amount = total * 0.22
+        total_with_tax = total * 1.22
+        withholding_amount = total * 0.2
+        total_with_withholding = total_with_tax - withholding_amount
+        cls.real_client_invoice_data = (total, tax_amount, total_with_tax, withholding_amount, total_with_withholding)
+
+        cls.withholding_tax_invoice._post()
+        cls.pension_fund_tax_invoice._post()
+
+    def test_withholding_tax_constraints(self):
+        with self.assertRaises(ValidationError):
+            self.withholding_sale_tax.amount = 10
+        with self.assertRaises(ValidationError):
+            self.withholding_sale_tax.l10n_it_withholding_type = False
+        with self.assertRaises(ValidationError):
+            self.withholding_sale_tax.l10n_it_withholding_reason = False
+        with self.assertRaises(ValidationError):
+            self.company.account_sale_tax_id.l10n_it_withholding_type = "RT02"
+
+    def test_withholding_taxes_export(self):
+        """
+            Invoice
+            -------------------------------------------------------------
+            Ordinary accounting service for the year               350.00
+            Balance deposit for the past year                      300.00
+            Ordinary accounting service for the trimester           50.00
+            Electronic invoices management                          50.00
+            -------------------------------------------------------------
+            Total untaxed:                                         750.00
+            Withholding:     20% of Untaxed Amount                -150.00
+            VAT:             22% of Untaxed Amount                 165.00
+            Document total:  Untaxed Amount + VAT                  915.00
+            Payment amount:  Document total - Withholding          765.00
+        """
+        invoice_etree = etree.fromstring(self.withholding_tax_invoice._export_as_xml())
+        total, tax_amount, total_with_tax, withholding_amount, total_with_withholding = self.real_client_invoice_data
+        expected_etree = self.with_applied_xpath(
```

Comment:
A lot of parameters, so do we put in a separate method so we can reuse it or do we hard-code?